base/src/main/java/io/spine/validate/FieldValidator.java

```diff
@@ -169,14 +170,34 @@ boolean fieldValueNotSet() {
         return result;
     }
 
+    private void canBeRequired() {
+        FieldValue desiredValue = fieldValueChange.newValue();
+        boolean requiredSetExplicitly = desiredValue.option(OptionsProto.required)
+                                                    .isExplicitlySet();
+        if (requiredSetExplicitly && !requiredAllowed()) {
+            log().warn("A field %s of type that can't be required was declared as" +
+                               "(required) = true. See required option documentation in " +
+                               "the options.proto file.", field().name());
+        }
+    }
+
+    /**
+     * Checks whether a field can be {@code (required) = true}.
+     *
+     * Validators can define that for their respective types
+     *
+     * @return whether a field that is being validated can be {@code required}
+     */
+    protected abstract boolean requiredAllowed();
```

Comment:
This method name is complicated. It uses 2 past form verbs. Please think of a better name.